rustls/src/client/hs.rs

```diff
@@ -420,13 +424,14 @@ fn emit_client_hello_for_retry(
     let next = ExpectServerHello {
         handshake,
         randoms,
+        using_ems,
         hello,
         early_key_schedule,
         sent_tls13_fake_ccs,
     };
 
     Ok(if support_tls13 && retryreq.is_none() {
-        Box::new(ExpectServerHelloOrHelloRetryRequest { next, extra_exts })
+        Box::new(ExpectServerHelloOrHelloRetryRequest { next, using_ems, extra_exts })
```

Comment:
Do we actually need to duplicate this across `ExpectServerHello` and `ExpectServerHelloOrHelloRetryRequest` (which contains `ExpectServerHello`)?

Reply:
Nope. Fixed.